src/components/ListItem.js

```diff
@@ -0,0 +1,50 @@
+import { doc, updateDoc } from 'firebase/firestore';
+import { db } from '../lib/firebase';
+import { useEffect, useState } from 'react';
+
+const oneDayInSeconds = 86400;
+
+const style = {
+       listStyleType: 'none',
+       textAlign: 'left'
+}
+const ListItem = ({ itemData }) => {
+    const [checked, setChecked] = useState(itemData.lastPurchased !== null)
+
+    const wasPurchasedWithin24Hours = () => {
+        if (itemData.lastPurchased === null) {
+            return false
+        }
+        return (Math.floor(Date.now() / 1000 ) - itemData.lastPurchased.seconds) <= oneDayInSeconds
+    }
+
+    useEffect(() => {
+        if (itemData.lastPurchased === null) {
+            return false
+        }
+           
+        if ((Math.floor(Date.now() / 1000 ) - itemData.lastPurchased.seconds) >= oneDayInSeconds) {
+            setChecked(false)
+        } else {
+            setChecked(true)
+        }
+      }, [itemData]);
+
+    const handleChange = () => {
+        const docRef = doc(db, "Lists", itemData.id)
+        if (itemData.lastPurchased === null || (Math.floor(Date.now() / 1000 ) - itemData.lastPurchased.seconds) >= oneDayInSeconds)  {
+            updateDoc(docRef, {lastPurchased: new Date()});
+            setChecked(true)
+        } else {
+            setChecked(false)
+        }
```

Comment:
same here! I guess we don't need to set the checked state to false. there is not an option to uncheck an item? the only way to uncheck an item is done automatically by useEffect when 24 hours have passed, right?